<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Calculating an Individual&#39;s Genetic Risk by epxlp</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Calculating an Individual&#39;s Genetic Risk</h1>
      <h2 class="project-tagline">from association to prediction</h2>
      <a href="https://github.com/epxlp/individual_genetic_risk" class="btn">View on GitHub</a>
      <a href="https://github.com/epxlp/individual_genetic_risk/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/epxlp/individual_genetic_risk/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <p>This handout can be found here: <a href="https://epxlp.github.io/individual_genetic_risk/">https://epxlp.github.io/individual_genetic_risk/</a></p>

<p>There are many ways in which you can estimate an individual’s risk of a particular disease or predict their likely trait value. How you do this will depend on the trait, the circumstances of the individual and the reason for wanting to make such a prediction.</p>

<p>In today’s session we will demonstrate estimating risk and predicting trait values in a variety of settings using different types of information.</p>

<h1>
<a id="the-data" class="anchor" href="#the-data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>The data</h1>

<p>Because we have our genes from birth, we could use an individual’s genetic data to estimate their risk having a certain disease later in life. In practice you might be more interested in doing this for cancer or heart disease, but we’ll do this for a trait that you have data for: <strong>BMI/obesity</strong>.</p>

<h1>
<a id="recapping-association-results" class="anchor" href="#recapping-association-results" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Recapping association results</h1>

<p>In the GWAS exercise in unit 2 you found a number of genetic variants associated with BMI. 
Let’s start by reminded ourselves of the association evidence between being BMI and the associated SNPs.</p>

<table>
<thead>
<tr>
<th>SNP</th>
<th>CHR</th>
<th>BP</th>
<th>A1</th>
<th>BETA</th>
<th>P</th>
<th>MAF</th>
<th>var_exp</th>
</tr>
</thead>
<tbody>
<tr>
<td>rs12748679</td>
<td>1</td>
<td>72644585</td>
<td>C</td>
<td>-0.598</td>
<td>1.3E-10</td>
<td>0.205</td>
<td>0.0038</td>
</tr>
<tr>
<td>rs630372</td>
<td>1</td>
<td>177885762</td>
<td>A</td>
<td>0.687</td>
<td>5.5E-15</td>
<td>0.236</td>
<td>0.0055</td>
</tr>
<tr>
<td>rs2867125</td>
<td>2</td>
<td>622827</td>
<td>T</td>
<td>-0.606</td>
<td>1.1E-09</td>
<td>0.168</td>
<td>0.0033</td>
</tr>
<tr>
<td>rs13130484</td>
<td>4</td>
<td>45175691</td>
<td>T</td>
<td>0.482</td>
<td>2.2E-10</td>
<td>0.434</td>
<td>0.0037</td>
</tr>
<tr>
<td>rs2920930</td>
<td>8</td>
<td>76733973</td>
<td>T</td>
<td>-0.470</td>
<td>1.1E-08</td>
<td>0.296</td>
<td>0.0030</td>
</tr>
<tr>
<td>rs8050136</td>
<td>16</td>
<td>53816275</td>
<td>A</td>
<td>0.594</td>
<td>9.5E-15</td>
<td>0.395</td>
<td>0.0055</td>
</tr>
<tr>
<td>rs12970134</td>
<td>18</td>
<td>57884750</td>
<td>A</td>
<td>0.950</td>
<td>4.0E-29</td>
<td>0.267</td>
<td>0.0114</td>
</tr>
</tbody>
</table>

<p><br><br></p>

<blockquote>
<p><strong>Task: Examine the strength of the evidence for the association (P-value), the magnitude of the association (beta/OR) and the variance in BMI explained for each of these SNPs, that you estimated in unit 2.<br>
How convinced are you that these variants are associated with BMI/being overweight?</strong></p>
</blockquote>

<p><br><br>
Some of these are very strong associations. p=3e-29 is extremely strong evidence of an association. Despite this the effect sizes and the variance explained are quite small {elaborate}. In unit 2 you estimated that the total variance explained for these 7 SNPs was 4%. </p>

<p><br><br></p>

<blockquote>
<p><strong>Question: These variants are all associated with being overweight, but what do you think determines how good a variant is for prediction?</strong></p>
</blockquote>

<p><br><br>
<strong>BUT – these only tell us how useful a genotype is ‘on average’ in the population. How useful a genotype is for prediction for an individual also depends on the individual’s genotype.
Some variants might have very large effect, but be extraordinarily rare. This variant would have a low variance explained, but for someone who had this genotype, this would be very informative. Discuss this with a neighbour</strong></p>

<h1>
<a id="generating-a-genetic-risk-score" class="anchor" href="#generating-a-genetic-risk-score" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Generating a genetic risk score</h1>

<p>First we are going to generate a file that details the SNPs we want to use in the genetic risk score. Copy the following command and check that the snps_for_score.txt file is created.</p>

<pre><code>echo 'rs571312        A       1
rs630372        A       1
rs8050136       A       1
rs12748679      T       1
rs13130484      T       1
rs2867125       C       1
rs2920930       G       1' &gt;  ~/ibsc_unit2/data/snps_for_score.txt
</code></pre>

<p>Use this command to generate a risk score for each person in the sample:</p>

<pre><code>plink --bfile ~/ibsc_unit2/data/geno_qc --score ~/ibsc_unit2/data/snps_for_score.txt --out ~/ibsc_unit2/data/BMI_score
</code></pre>

<p>Now we will load the data into R</p>

<pre><code>score &lt;- read.table(“~/ibsc_unit2/data/BMI_score.profile”, header=T)
phen &lt;- read.table(“~/ibsc_unit2/data/phen_clean.txt”)
</code></pre>

<p>Run the following commands to investigate the distribution of the genetic risk score</p>

<pre><code>BMI_score &lt;- score$CNT2
hist(BMI_score)
hist(BMI_score[phen$BMIcat=="underweight" | phen$BMIcat=="healthy"], col=rgb(1,0,0,0.5), xlim=c(0,14), ylim=c(0,600), main = "BMI categories", xlab="genetic_score")
hist(BMI_score[phen$BMIcat=="obese"], col=rgb(0,0,1,0.5), add=T)
</code></pre>

<h1>
<a id="population-level-predictions" class="anchor" href="#population-level-predictions" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Population-level predictions</h1>

<p>What is the difference in obesity risk between those in the top 5% and bottom 5% of the score?</p>

<pre><code>quantile(BMI_score, c(.05, .95))
sum(phen$BMIcat=="obese" &amp; BMI_score&lt;5)/sum(BMI_score&lt;5)
sum(phen$BMIcat=="obese" &amp; BMI_score&gt;10)/sum(BMI_score&gt;10)
</code></pre>

<p>There seems to be quite a significant difference in risk. Those in the bottom 5% of the risk score have only a 30% risk of being obese, whilst those in the top 5% of the risk score have a 60% risk of being obese.</p>

<p><br><br></p>

<blockquote>
<p><strong>Question: Do you think it would be an economically effective strategy to target an intervention only at children that were in the top 5% of the risk score?<br>
Although this is unlikely to happen for an outcome such as BMI, think about if the outcome were cancer or Alzheimer’s disease.<br>
Discuss this with a neighbour</strong></p>
</blockquote>

<p><br><br>
<strong>As a genetic risk score increases there is a sliding scale of increased risk, but any health policies would likely have to define a threshold and this would have sensitivity and specificity implications.</strong></p>

<h1>
<a id="individual-level-prediction" class="anchor" href="#individual-level-prediction" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Individual level prediction</h1>

<p>There’s clearly a difference in the genetic_score distribution between healthy and underweight versus obese individuals. However, if I told you someone had a genetic score of 4, what would that tell you?</p>

<pre><code>table (BMI_score, phen$BMIcat)
summary(phen$BMI[BMI_score==4])
boxplot(phen$BMI~BMI_score)
</code></pre>

<p>Of 335 people in this category: 7% are underweight, 27% are healthy, 35% are overweight, 32% are obese. So you might say that they are quite likely to be overweight or obese, but the range of BMIs for people in this category is very wide (11.5 to 43.3), so it would be difficult to predict their BMI with any accuracy.</p>

<p>Compare the mean BMIs of people at the top and bottom of the central 90%</p>

<pre><code>quantile(BMI_score, c(.05, .95))
summary(phen$BMI[BMI_score==5])
summary(phen$BMI[BMI_score==10])
</code></pre>

<p>90% of people fall between BMI_scores of 5 and 10, the mean BMI difference between the top and bottom of this range is: 2.81kg/m2 BMI=30 versus BMI=27. </p>

<p><br><br></p>

<blockquote>
<p><strong>For the majority of people knowing your BMI score is very useful / somewhat useful / pretty useless (delete as applicable)</strong></p>
</blockquote>

<p>If we classed everyone with BMI_score &gt;10 as at risk of obesity. How often would we be right and wrong?</p>

<pre><code>obese &lt;- phen$BMIcat
levels(obese) &lt;- c("control", "obese", "control", "control")
risk &lt;- factor(BMI_score)
levels(risk) &lt;- c("&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&gt;10", "&gt;10", "&gt;10")
table(obese, risk)
</code></pre>

<p><br><br></p>

<blockquote>
<p><strong>Give the numbers in each category:<br>
given low risk prediction &amp; did not become obese:<br>
given low risk prediction &amp; did become obese:<br>
given high risk prediction &amp; did become obese:<br>
given high risk prediction &amp; did not become obese: 
<br><br> 
Question: what proportion of the time would you be right?</strong></p>
</blockquote>

<p><br><br>
You are only going to be right 57% of the time. So for an individual you can make a prediction, but it’s not much better than flipping a coin</p>

<p>What about if you took BMI_score&gt;=13 as the cut-off?</p>

<pre><code>risk &lt;- factor(BMI_score)
levels(risk) &lt;- c("&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&lt;=10", "&gt;10")
table(obese, risk)
</code></pre>

<p><br><br></p>

<blockquote>
<p><strong>Give the numbers in each category:<br>
given low risk prediction &amp; did not become obese:<br>
given low risk prediction &amp; did become obese:<br>
given high risk prediction &amp; did become obese:<br>
given high risk prediction &amp; did not become obese:</strong></p>
</blockquote>

<p><br><br>
In this dataset for the 2 people you predicted would be obese, you would be right, but you would have incorrectly told 3524 people that they were not in the high risk group.</p>

<p><br><br></p>

<blockquote>
<p><strong>Question: What proportion of people have a genetic risk score&gt;=13?<br>
Can we consider this a similar situation as the rare high penetrant mutations you were considering in the other part of this practical?</strong></p>
</blockquote>

<h1>
<a id="published-examples" class="anchor" href="#published-examples" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Published examples</h1>

<p>Some studies have attempted to predict BMI from genotype (e.g. Speliotes 2010, Nature Genetics 42(11):937-948)</p>

<p><img src="https://github.com/epxlp/individual_genetic_risk/blob/gh-pages/fig1.png" alt="hover">
<img src="fig1.png"></p>

<p>Fig a. Mean BMI according to risk score - Note there is an increase of ~3kg/m2 between the top and bottom of the genetic risk score
Fig b. Displays the ROC curve for 2 prediction models. Solid line includes age, age2 and sex (AUC=0.515), dashed line includes age, age2, sex and 32 BMI SNPs (AUC=0.575).</p>

<p><br><br></p>

<blockquote>
<p><strong>Question: Does including the SNPs improve the prediction of obesity?</strong></p>
</blockquote>

<p><br><br>
Another study Morandi PloS ONE 2012, 7(11):e49919 compared genetic prediction to prediction using traditional risk factors.</p>

<img src=fig2.png>

<p>Turquoise= genetic risk score, Red= traditional risk factors, dark blue= combined traditional risk factprs and genetic risk score.
Integrated discrimination improvement (IDI) = 0.5%, shows the possible accuracy improvement associated with adding the genetic score to the traditional risk factors alone. </p>

<p><br><br></p>

<blockquote>
<p><strong>Question: Is there any improvement including SNPs in the prediction?</strong></p>
</blockquote>

<p><br><br></p>

<p>Look up this study now and see which traditional risk factors were included in the model.</p>

<p><br><br></p>

<blockquote>
<p><strong>Why does a prediction that includes ‘parental BMI’ do just as well as one that includes SNPs?</strong></p>
</blockquote>

<p><br><br></p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/epxlp/individual_genetic_risk">Calculating an Individual&#39;s Genetic Risk</a> is maintained by <a href="https://github.com/epxlp">epxlp</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
